plant/plants/api/views.py

```diff
@@ -28,6 +44,47 @@ def get_queryset(self):
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    @action(detail=True, methods=["GET"])
+    def plant_stats(self, request, pk=None):
+        start_date = request.query_params.get("start_date")
+        end_date = request.query_params.get("end_date")
+        plant = get_object_or_404(Plant, pk=pk)
+        the_most_active_users = (
+            plant.waterings.values("user")
+            .order_by("user")
+            .annotate(total_litres=Sum("litres"))[:3]
+        )
+        average_watering_per_month = plant.waterings.values(
+            month=F("watering_date__month")
+        ).annotate(average_litres=Avg("litres"))
+
+        print(start_date, end_date)
+
+        if start_date and end_date:
+            waterings_count = plant.waterings.filter(
+                watering_date__range=(start_date, end_date)
+            ).count()
+        else:
+            waterings_count = plant.waterings.filter(
+                watering_date__lte=datetime.now() - timedelta(days=30)
+            ).count()
+
+        average_serializer = AverageWateringPerMonthSerializer(
+            average_watering_per_month, many=True
+        )
+        active_user_serializer = TheMostActiveUsersSerializer(
+            the_most_active_users, many=True
+        )
+
+        return Response(
+            {
+                "active_user": active_user_serializer.data,
```

Comment:
You could consider adding swagger for this endpoint using extend_schema decorator as this response will not be captured by drf-spectacular.
Also we could discuss about adding another serializer that encapsulates these 3 fields (not sure if it's that much of improvement but I think swagger will be resolved automatically then), so you can decide